### ISSUE TITLE
dnsmasq not resolving docker cloud links

### ISSUE THREAD

**Issue by robbydooo** [Mashape/docker-kong#22]
Hi All,

I am trying to get Kong properly working on Tutum/Docker Cloud (as it is now called).

https://cloud.docker.com

The issue is that kong always fails when using docker links on Tutum/DockerCloud.

It appears that DNSmasq is not using the search element of the resolve.conf to add the FQDN to the request and as such it just fails.

So if I set in kong.yml :

```
contact_points:
    - "cassandra:9042"
```

Kong will fail to start unable to connect to cassandra. It should resolve to casasndra..<container-id>..local.dockerapp.io

Example of resolve.conf on docker cloud:

```
search .<container-id>..local.dockerapp.io
nameserver 8.8.8.8
nameserver 8.8.4.4
```

This issue also prevents you using microservices behing kong via docker links as they do not resolve the same way as cassandra does not.

E.g http://chatservice should resolve to http://chatservice.<container-id>.local.dockerapp.io

container-id changes each time you redeploy or bring new containers up so you can not hard code this, the DNS for .<container-id> is only active for as long as the container is.

How can I get kong/dnsmasq to properly use the search option in resolve.conf?

If we can solve this issue then I will no longer have to  host a custom version of Kong for Docker Cloud

https://github.com/holidaywatchdog/docker-cloud-kong

I should mention that cassandra resolves correctly from within the docker container just not via dnsmasq

```
root@api-gateway-1 /]# ping cassandra
PING cassandra.a72c1fd7-97e3-423b-ac69-d02a5dee5d8e.local.dockerapp.io (10.7.0.2
2) 56(84) bytes of data.
64 bytes from 10.7.0.22: icmp_seq=1 ttl=64 time=1.79 ms
```

The only short term work around I have for this is to use a dummy container in the stack and link it to all the services you need to use with kong. Then use that containers FQDN to reference the services. As long as this container never goes down the DNS should work.

**Comment by begizi** [comment]
Has there been any progress on this? I've been trying to deploy kong on docker-cloud with postgres without any luck.

**Comment by robermorales** [comment]
Any update on this?

**Comment by thibaultcha** [comment]
As of 0.10, Kong does not use dnsmasq anymore but its own, embedded DNS resolver. And as of 0.11 (of which our first release candidate is currently available for testing) implements support for `search` and `ndots` options.

Expect this to be released when 0.11 will be officially released (a few weeks from now). I will be closing this in the meantime as it is currently not applicable/already being worked on. Please open a new issue if you are stumbling upon this as of 0.11. Thanks!
